commons/commons-model-extended-serialization/src/main/kotlin/uk/ac/ebi/extended/serialization/deserializers/ExtFileDeserializer.kt

```diff
@@ -27,11 +33,31 @@ class ExtFileDeserializer : JsonDeserializer<ExtFile>() {
 
         return when (val type = ExtType.valueOf(node.getNode<TextNode>(EXT_TYPE).textValue())) {
             is ExtType.NfsFile -> nfsFile(node, mapper)
-            is ExtType.FireFile -> TODO()
+            is ExtType.FireFile -> fireFile(node, mapper)
+            is ExtType.FireDirectory -> fireDirectory(node, mapper)
             else -> throw InvalidExtTypeException(type.type)
         }
     }
 
+    private fun fireDirectory(node: JsonNode, mapper: ObjectMapper): FireDirectory {
+        return FireDirectory(
+            fileName = node.getNode<TextNode>(FILE_NAME).textValue(),
+            md5 = "md5",
+            size = node.getNode<IntNode>(FILE_SIZE).longValue(),
+            attributes = mapper.convertList(node.findNode(ATTRIBUTES))
+        )
+    }
+
+    private fun fireFile(node: JsonNode, mapper: ObjectMapper): FireFile {
+        return FireFile(
+            fileName = node.getNode<TextNode>(FILE_NAME).textValue(),
+            fireId = node.getNode<TextNode>(FILE_FIRE_ID).textValue(),
+            md5 = "md5",
```

Comment:
lets assume md5 is provided. (sounds not realistic for submission tool but it is for migration)